cmake/herbert_FindMatlab.cmake

```diff
@@ -17,26 +17,63 @@ the path to the directory containing Matlab's dynamically linked libraries
 ``Matlab_UT_LIBRARY``
 the path to the Matlab UT library
 
-``Matlab_VERSION``
+``Matlab_RELEASE``
 the Matlab release e.g. R2019b
 
 See the FindMatlab.cmake documentation for other variables defined by this
 module. You'll find the file bundled with your CMake installation.
 
 #]=======================================================================]
-find_package(Matlab REQUIRED COMPONENTS MAIN_PROGRAM MEX_COMPILER)
+include(MatlabHelpers)
+
+root_dir_changed(_root_changed)
+release_changed(_release_changed)
+if(_root_changed AND NOT _release_changed)
+    # If Matlab_ROOT_DIR has been changed but not Matlab_RELEASE, discard the
+    # old Matlab_RELEASE in favour of the ROOT_DIR
+    unset(Matlab_RELEASE CACHE)
+elseif(_release_changed AND NOT _root_changed)
+    # If Matlab_RELEASE has been changed but not Matlab_ROOT_DIR, discard the
+    # old Matlab_ROOT_DIR in favour of the Matlab_RELEASE
+    unset(Matlab_ROOT_DIR CACHE)
+endif()
+
+if("${Matlab_RELEASE}" STREQUAL "")  # specific version not required
+    find_package(Matlab COMPONENTS MAIN_PROGRAM MEX_COMPILER)
+else()
+    matlab_get_version_from_release_name("${Matlab_RELEASE}" _version)
+    find_package(Matlab EXACT ${_version} COMPONENTS MAIN_PROGRAM MEX_COMPILER)
+endif()
+
+# Get the release of the Matlab that's been found
+matlab_get_release_at_path("${Matlab_ROOT_DIR}" _found_release)
+
+# Set local cached versions of variables so changes on next run can be tracked
+set(_CACHED_Matlab_RELEASE "${Matlab_RELEASE}" CACHE INTERNAL "")
+set(_CACHED_MATLAB_ROOT_DIR "${Matlab_ROOT_DIR}" CACHE INTERNAL "")
+
+if(${Matlab_FOUND})
+    set_matlab_release("${_found_release}")
+else()
+    set_matlab_release("${_CACHED_Matlab_RELEASE}")
+endif()
+
+# Throw error if the Matlab found does not match the Matlab requested by Matlab_RELEASE
+if(NOT "${_found_release}" STREQUAL "${Matlab_RELEASE}"
+        AND NOT "${_INPUTTED_MATLAB_RELEASE}" STREQUAL "")
```

Comment:
`_INPUTTED_MATLAB_RELEASE` is never set of cleared, so I'm not sure how this condition can ever be hit

Reply:
it looks like there is a piece of logic that we were depending on earlier that has now gone